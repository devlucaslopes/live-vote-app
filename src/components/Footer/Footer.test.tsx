import React from 'react'
import { render, screen } from '@testing-library/react'

import { Footer } from '.'

describe('<Footer />', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(<Footer />)

    expect(screen.getByRole('contentinfo')).toBeInTheDocument()
    expect(screen.getByText(/criado por @devlucaslopes/i)).toBeInTheDocument()
    expect(screen.getByRole('link')).toHaveProperty(
      'href',
      'http://twitch.tv/devlucaslopes'
    )
  })

  it('should center text', () => {
    render(<Footer />)

    expect(screen.getByTestId('footer-content')).toHaveClass(
      'flex justify-center'
    )
  })
})
