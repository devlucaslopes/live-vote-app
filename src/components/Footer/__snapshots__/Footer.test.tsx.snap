// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<div>
  <footer
    class="h-8 py-1 bg-pink-500"
  >
    <div
      class="flex justify-center"
      data-testid="footer-content"
    >
      <a
        class="text-slate-900"
        href="http://twitch.tv/devlucaslopes"
        rel="noreferrer"
        target="_blank"
      >
        Criado por @devlucaslopes
      </a>
    </div>
  </footer>
</div>
`;
